website/content/docs/configuration/index.mdx

```diff
@@ -97,9 +97,12 @@ to specify where the configuration is.
   the read cache used by the physical storage subsystem. This will very
   significantly impact performance.
 
-- `disable_mlock` `(bool: false)` – Disables the server from executing the
-  `mlock` syscall. `mlock` prevents memory from being swapped to disk. Disabling
-  `mlock` is not recommended unless using [integrated storage](/vault/docs/internals/integrated-storage).
+- `disable_mlock` `(bool: false)` – Required to be explicitly set
```

Comment:
Could we also add a brief note describing the tradeoff in https://developer.hashicorp.com/vault/docs/concepts/production-hardening ? I think the exclusions in  https://developer.hashicorp.com/vault/docs/internals/security and the opt-in nature is sufficient, but curious if you think we should make a small change there too.

Reply:
Those are good callouts, thank you. I'll add something to production-hardening about the tradeoffs and read over the security model and let you know my thoughts.

Reply:
Hi @mickael-hc ! I have updated the production hardening section to describe the tradeoff between `disable_mlock` config options. I read over the threat model I believe that that this particular tradeoff would fall under the items that are _not_ part of the Vault threat model, right? If that's the case, we may want to call out  "inspecting swap" as explicitly not a part of our threat model. What do you think?

Reply:
I agree, I think the production hardening guide (as you've already done) is sufficient here, thanks!